### DISCUSSION TITLE
`brew test gdal` fails at least with the arm64 build

### DISCUSSION THREAD

**Discussion by elbamos** [Homebrew/discussions#948] [Everyday usage]
I'm having trouble compiling against gdal (trying to compile rgdal and sf, R packages). I've been able to get them to compile when I build gdal myself from source, and trying to diagnose what's going wrong when I try to compile against the homebrew build. 

To try to track it down I ran `brew test gdal`, which fails as follows:

```
 % brew test gdal
==> Testing gdal
==> /opt/homebrew/Cellar/gdal/3.2.1_2/bin/gdalinfo --formats
Last 15 lines from /Users/amos.elberg/Library/Logs/Homebrew/gdal/test.01.gdalinfo:
2021-02-26 11:46:12 -0800

/opt/homebrew/Cellar/gdal/3.2.1_2/bin/gdalinfo
--formats

Error: gdal: failed
An exception occurred within a child process:
  BuildError: Failed executing: /opt/homebrew/Cellar/gdal/3.2.1_2/bin/gdalinfo --formats
/opt/homebrew/Library/Homebrew/formula.rb:2112:in `block in system'
/opt/homebrew/Library/Homebrew/formula.rb:2048:in `open'
/opt/homebrew/Library/Homebrew/formula.rb:2048:in `system'
/opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/gdal.rb:170:in `block in <class:Gdal>'
/opt/homebrew/Library/Homebrew/formula.rb:1915:in `block (3 levels) in run_test'
/opt/homebrew/Library/Homebrew/utils.rb:550:in `with_env'
/opt/homebrew/Library/Homebrew/formula.rb:1914:in `block (2 levels) in run_test'
/opt/homebrew/Library/Homebrew/formula.rb:900:in `with_logging'
/opt/homebrew/Library/Homebrew/formula.rb:1913:in `block in run_test'
/opt/homebrew/Library/Homebrew/mktemp.rb:63:in `block in run'
/opt/homebrew/Library/Homebrew/mktemp.rb:63:in `chdir'
/opt/homebrew/Library/Homebrew/mktemp.rb:63:in `run'
/opt/homebrew/Library/Homebrew/formula.rb:2161:in `mktemp'
/opt/homebrew/Library/Homebrew/formula.rb:1907:in `run_test'
/opt/homebrew/Library/Homebrew/test.rb:43:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:93:in `block in timeout'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `block in catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:108:in `timeout'
/opt/homebrew/Library/Homebrew/test.rb:42:in `<main>'
```

It appears that /opt/homebrew/Cellar/gdal/3.2.1_2/bin/gdalinfo fails whenever run:

```
% /opt/homebrew/Cellar/gdal/3.2.1_2/bin/gdalinfo
zsh: killed     /opt/homebrew/Cellar/gdal/3.2.1_2/bin/gdalinfo
```

This is the same behavior (killed before execution of any code) that I'm seeing from the `configure` scripts for `rgdal` and `sf`, where they attempt to build a simple executable that links against gdal, compilation is fine, but the executable is killed on launch before executing any code.

**Comment by carlocab** [comment]
Try doing
```
brew reinstall $(brew deps gdal)
```

**Reply by elbamos** [reply]
Wow that did clear it out, thanks! I saw package after package that had, for some reason, not been upgraded by `brew upgrade`. I had no idea brew was capable of getting itself into such an inconsistent state.

**Reply by carlocab** [reply]
That command shouldn't have upgraded anything; it should've only reinstalled a bunch of formulae. If it actually did upgrade something then something might be wrong with your setup.

**Reply by elbamos** [reply]
That command didnt upgrade anything. What it did, was that for a slew of packages, the reinstallation failed on a permission error. The cause of the permission error was that reinstall was trying to replace files that didn't exist, because it was trying to reinstall one version but the version then-currently installed was actually a prior version. I had to upgrade each of those packages (it was perhaps 8) manually. The reinstall then completed without error and the builds were then able to execute.
